packages/ssx-sdk/src/modules/common.ts

```diff
@@ -0,0 +1,12 @@
+export interface ResourceError {
+  type: string;
+  message: string;
+  data?: any;
+}
+
+export interface ResourceResult {
+  success: boolean;
```

Comment:
I'm still not convinced that this pattern is necessary - why not `throw ResourceError`/`return data`?